prometheus-client/src/Prometheus/Metric.hs

```diff
@@ -45,10 +45,7 @@ data SampleGroup = SampleGroup Info SampleType [Sample]
 -- of a handle value and a collect method. The handle value is typically a new
 -- type wrapped value that provides access to the internal state of the metric.
 -- The collect method samples the current value of the metric.
-data Metric s = Metric {
-        handle  :: s
-    ,   collect :: IO [SampleGroup]
-    }
+newtype Metric s = Metric { construct :: IO (s, IO [SampleGroup]) }
```

Comment:
Please document `construct`. Maybe worth updating the main `Metric` docs too?

Reply:
I think the main `Metric` doc is fine, it still does the same thing. I'll document `construct` though.